### PR TITLE
CASSANDRA-18839 trunk Catch SSLException

### PR DESCRIPTION
**Pull request by smiklosovic**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

**Review by maedhroz — DISMISSED** [review]

File: src/java/org/apache/cassandra/transport/ExceptionHandlers.java [hunk 1]
```diff
@@ -133,6 +136,15 @@ else if (Throwables.anyCauseMatches(cause, t -> t instanceof Errors.NativeIoExce
...
         }
+        else if (Throwables.anyCauseMatches(cause, t -> t instanceof SSLHandshakeException))
+        {
+            ClientMetrics.instance.markSSLHandshakeException();
+            NoSpamLogger.log(logger, NoSpamLogger.Level.WARN, 1, TimeUnit.MINUTES, "SSLHandshakeException in client networking with peer {} {}", clientAddress, cause.getMessage());
+        }
+        else if (Throwables.anyCauseMatches(cause, t -> t instanceof SSLException))
+        {
+            NoSpamLogger.log(logger, NoSpamLogger.Level.WARN, 1, TimeUnit.MINUTES, "SSLException in client networking with peer {} {}", clientAddress, cause.getMessage());
+        }
```
Comment (maedhroz) [inline src/java/org/apache/cassandra/transport/ExceptionHandlers.java:147]: The only thing that worries me a little bit here is what happens when we have overloads, protocol exceptions, and SSL problems all occurring at the same time. It's not that expensive to create a new logger to wrap, specifically for SSL exceptions. Any objection to doing that?

Comment (smiklosovic) [reply src/java/org/apache/cassandra/transport/ExceptionHandlers.java:147]: I am not sure I am getting your comment right ... The way how NoSpamLogger seems to work is that it will not repeat same message all over again during some period. It does not mean it will not repeat any message if some was already logged.

Example: lets say we have messages to log, "msg1" and "msg2". What NoSpamLogger does is that it will log "msg1" and then "msg2" but if you ever send any other "msg2" to it during 1 minute, it will not log that. But that does not mean that it will not log msg1 if 1 minute passed from its last log. 

In other words, these periods of 1 minute are _per message_, not _per logger_. Logger via which we log these messages might be same.

**Review by maedhroz — APPROVED** [review]

File: src/java/org/apache/cassandra/transport/ExceptionHandlers.java (hunk 1 above)
Comment (maedhroz) [reply src/java/org/apache/cassandra/transport/ExceptionHandlers.java:147]: You are 100% correct. I only looked at `NoSpamLogger` acquisition and ignored `NoSpamLogStatement` acquisition.
